docs/Event-Tracking-and-Schemas.md

```diff
@@ -0,0 +1,80 @@
+# Event Tracking and Schemas
+
+In order to understand how well (or badly) a banner is performing we track donor events that are sent to external services. Common actions
+that are tracked include:
+
+- Banner was seen
+- Banner was not shown
+- Banner was submitted
+- A feature was interacted with
+
+If a new feature is added the performance of that feature is tested by adding it to a VAR banner and comparing it to a CTRL banner. We use
+tracking events to see if the feature is interacted with plus the overall performance of each banner is compared to see if there are changes
+in donor behaviour.
+
+## Domains
+
+We host banners in 2 different places, on wikipedia.org (ORG) and on wikipedia.de (DE). Banners hosted on DE use our self-hosted Matomo
+tracking, and banners hosted on ORG use the Wikimedia Event Platform.
+
+### Wikimedia Event Platform
+
+This is a very powerful tracking tool that is used to track all events that happen on wikipedia.org.
+
+### Matomo
+
+This is an open source analytics tool. It is very basic compared to the Event Platform meaning we can't track as much as we do on the ORG
+banners, but it still works well enough for our purposes.  
+
+## Event Mapping
+
+This separation requires two separate implementations, one for each external service. 
+
+Banners hosted on either domain fire the same internal events. These are bubbled up to the entry points, and they are then handled by the
+appropriate tracking code and sent to the external service.
+
+In the banner folders you will find an `event_map.ts` file that handles mapping the internal events to external ones.
```

Comment:
I remember us talking about that the shiny new event tracking system will make the mapping obsolete, which is also why we have those "temporary" `event_map.ts` files and why we put the events we send to event platform into the `LegacyEventTracking` folder.

As far as I understand, it would be a huge hassle to recreate our domain events as fragments and would "block externally" any banner ticket that tries to introduce a new tracking metric. Is that correct?

If it is correct, I suggest we centralize the mapping a bit more to avoid duplication and drift in the `event_map.ts` files and dropping the "Legacy" moniker.